File: Student Grade Calculator project/grade_calculator_flutter/grade_calculator_flutter/lib/core/theme.dart
// lib/core/theme.dart
// ─────────────────────────────────────────────────────────────────────────────
// Central theme definition — dark + light, Material 3.
// ─────────────────────────────────────────────────────────────────────────────

import 'package:flutter/material.dart';
import 'package:google_fonts/google_fonts.dart';

class AppTheme {
  AppTheme._();

  // Brand palette
  static const Color primary        = Color(0xFF4F6AF5);
  static const Color primaryDark    = Color(0xFF3A52D4);
  static const Color accent         = Color(0xFF00D4B4);
  static const Color danger         = Color(0xFFFF5252);
  static const Color warning        = Color(0xFFFFAB40);
  static const Color success        = Color(0xFF66BB6A);
  static const Color surface        = Color(0xFF1A1D2E);
  static const Color surfaceVariant = Color(0xFF242742);
  static const Color cardBg         = Color(0xFF2A2D45);
  static const Color border         = Color(0xFF3A3D5C);
  static const Color textPrimary    = Color(0xFFF0F2FF);
  static const Color textSecondary  = Color(0xFF9B9FC4);

  // Grade colours
  static const Map<String, Color> gradeColours = {
    'A'  : Color(0xFF66BB6A),
    'B+' : Color(0xFF26C6DA),
    'B'  : Color(0xFF42A5F5),
    'C+' : Color(0xFFAB47BC),
    'C'  : Color(0xFF7E57C2),
    'D+' : Color(0xFFFFCA28),
    'D'  : Color(0xFFFFA726),
    'F'  : Color(0xFFEF5350),
  };

  static Color gradeColor(String letter) =>
      gradeColours[letter] ?? textSecondary;

  static ThemeData get dark {
    final base = ThemeData.dark(useMaterial3: true);
    return base.copyWith(
      colorScheme: ColorScheme.dark(
        primary       : primary,
        secondary     : accent,
        surface       : surface,
        error         : danger,
        onPrimary     : Colors.white,
        onSurface     : textPrimary,
      ),
      scaffoldBackgroundColor: surface,
      textTheme: GoogleFonts.interTextTheme(base.textTheme).apply(
        bodyColor   : textPrimary,
        displayColor: textPrimary,
      ),
      cardTheme: CardTheme(
        color        : cardBg,
        elevation    : 0,
        shape        : RoundedRectangleBorder(
          borderRadius: BorderRadius.circular(16),
          side        : const BorderSide(color: border, width: 1),
        ),
      ),
      elevatedButtonTheme: ElevatedButtonThemeData(
        style: ElevatedButton.styleFrom(
          backgroundColor: primary,
          foregroundColor: Colors.white,
          padding: const EdgeInsets.symmetric(horizontal: 24, vertical: 14),
          shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(12)),
          textStyle: GoogleFonts.inter(fontWeight: FontWeight.w600, fontSize: 14),
        ),
      ),
      inputDecorationTheme: InputDecorationTheme(
        filled          : true,
        fillColor       : surfaceVariant,
        border          : OutlineInputBorder(
          borderRadius: BorderRadius.circular(10),
          borderSide  : const BorderSide(color: border),
        ),
        enabledBorder   : OutlineInputBorder(
          borderRadius: BorderRadius.circular(10),
          borderSide  : const BorderSide(color: border),
        ),
        focusedBorder   : OutlineInputBorder(
          borderRadius: BorderRadius.circular(10),
          borderSide  : const BorderSide(color: primary, width: 2),
        ),
        labelStyle : const TextStyle(color: textSecondary),
        hintStyle  : const TextStyle(color: textSecondary),
      ),
      chipTheme: ChipThemeData(
        backgroundColor : surfaceVariant,
        selectedColor   : primary.withOpacity(0.3),
        labelStyle      : GoogleFonts.inter(fontSize: 13, color: textPrimary),
        side            : const BorderSide(color: border),
        shape           : RoundedRectangleBorder(borderRadius: BorderRadius.circular(8)),
      ),
      dividerTheme: const DividerThemeData(color: border, thickness: 1),
      scrollbarTheme: ScrollbarThemeData(
        thumbColor: WidgetStateProperty.all(primary.withOpacity(0.4)),
      ),
    );
  }
}

File: gradecalc_final/src/main/kotlin/ui/Theme.kt
// src/main/kotlin/ui/Theme.kt
package com.gradecalc.ui

import androidx.compose.material3.darkColorScheme
import androidx.compose.ui.graphics.Color

object AppColors {
    val Primary        = Color(0xFF4F6AF5)
    val Accent         = Color(0xFF00D4B4)
    val Danger         = Color(0xFFEF5350)
    val Warning        = Color(0xFFFFAB40)
    val Success        = Color(0xFF66BB6A)
    val Surface        = Color(0xFF1A1D2E)
    val SurfaceVariant = Color(0xFF242742)
    val CardBg         = Color(0xFF2A2D45)
    val Border         = Color(0xFF3A3D5C)
    val TextPrimary    = Color(0xFFF0F2FF)
    val TextSecondary  = Color(0xFF9B9FC4)

    private val gradeMap = mapOf(
        "A"  to Color(0xFF66BB6A),
        "B+" to Color(0xFF26C6DA),
        "B"  to Color(0xFF42A5F5),
        "C+" to Color(0xFFAB47BC),
        "C"  to Color(0xFF7E57C2),
        "D+" to Color(0xFFFFCA28),
        "D"  to Color(0xFFFFA726),
        "F"  to Color(0xFFEF5350)
    )

    fun forGrade(letter: String): Color  = gradeMap[letter] ?: TextSecondary
    fun forScore(score: Double): Color   = when {
        score >= 80 -> Success
        score >= 50 -> Accent
        score >= 40 -> Warning
        else        -> Danger
    }
    fun forFormat(label: String): Color  = when {
        label.contains("xlsx", true) -> Color(0xFF217346)
        label.contains("pdf",  true) -> Color(0xFFDC3545)
        label.contains("xml",  true) -> Color(0xFFFF6B35)
        label.contains("docx", true) -> Color(0xFF2B5899)
        else                         -> Primary
    }

    val colorScheme = darkColorScheme(
        primary   = Primary,
        secondary = Accent,
        surface   = Surface,
        error     = Danger,
        onSurface = TextPrimary
    )
}
